### ISSUE TITLE
Scheduler fails to notify compute node of cancellations

### ISSUE THREAD

**Issue by rossjones** [bacalhau-project/bacalhau#2035]
After submitting a cancellation request, the scheduler is unable to notify the compute node of the cancellation.

```
DBG pkg/requester/scheduler.go:118 > Requester node QmU8zYxvTfRHpW9SKVBfiEEntwPbFdnzLwU6qL4CQFJQry 
    received CancelJob for job: 3372ab35-bcdd-4835-8578-45f32908a6fb with reason Canceled at user request 
    [NodeID:QmU8zYxv]
INF pkg/requester/scheduler.go:672 > stopping shard 3372ab35-bcdd-4835-8578-45f32908a6fb:0 
    because the user requested it [NodeID:QmU8zYxv]
DBG pkg/requester/scheduler.go:334 > Requester node QmU8zYxvTfRHpW9SKVBfiEEntwPbFdnzLwU6qL4CQFJQry 
    responding with Cancel for bid: e-1df6703f-7405-42f0-b972-4bff90e98596 [NodeID:QmU8zYxv]
ERR pkg/requester/scheduler.go:346 > failed to notify cancellation for execution: 
    e-1df6703f-7405-42f0-b972-4bff90e98596 
    error="compute.CancelExecutionRequest: failed to open stream to peer 
    QmSsUZwS6tbDu1f33GHGBKyydmUo2XTLbMP3LTuy5NSCss: context canceled" [NodeID:QmU8zYxv]
```

**Comment by rossjones** [comment]
This is raised at https://github.com/bacalhau-project/bacalhau/blob/main/pkg/transport/bprotocol/compute_proxy.go#L129-L133

**Comment by rossjones** [comment]
This was fixed when DetachedContext moved in e82bd9eafb8935e4a8b744bc3cc55be5244fd208
